formats/html.ts

```diff
@@ -1,15 +1,64 @@
 import { FormatImporter } from "../format-importer";
-import { Notice, htmlToMarkdown, normalizePath } from "obsidian";
-import { pathToFilename } from '../util';
+import { Notice, Setting, TFile, TFolder, htmlToMarkdown, normalizePath, requestUrl } from "obsidian";
+import { pathToBasename, pathToFilename, sanitizeFileName, splitFilename } from '../util';
 import { ImportResult } from '../main';
+import { URL, fileURLToPath, pathToFileURL } from "url";
+import { readFile } from "fs/promises";
+import { getExtension, getType } from "mime/lite";
+import { disableFS, imageSize } from "image-size";
+import { fileTypeFromBuffer } from "file-type";
+import isSvg from "is-svg";
+
+disableFS(true);
 
 export class HtmlImporter extends FormatImporter {
+	attachments: Record<string, ReturnType<typeof this.downloadAttachment>> = {};
+
+	attachmentSizeLimit: number;
+	minimumImageSize: number;
+
 	init() {
 		this.addFileOrFolderChooserSetting('HTML (.htm .html)', ['htm', 'html']);
+		this.addAttatchmentSizeLimit(10);
+		this.addMinimumImageSize(65); // 65 so that 64×64 are excluded
 		this.addOutputLocationSetting('HTML');
 	}
 
-	async import(): Promise<void> {
+	addAttatchmentSizeLimit(defaultInMB: number) {
+		this.attachmentSizeLimit = defaultInMB * 10 ** 6;
+		new Setting(this.modal.contentEl)
+			.setName("Attachment size limit (MB)")
+			.setDesc("Set 0 to disable.")
+			.addText(text => text
```

Comment:
Did that.

But it only checks whether it can be converted using `Number` and doesn't allow you to set further constraints on the input, so most validation code in `onChange` is still there. I think the only benefit would be that mobile users would be shown the special numeric keyboard (if the plugin is ever supported on mobile).

I tried using `inputmode="numeric" pattern="(custom pattern here)"`, but I realized the pattern doesn't constrain the input but only marks the input as invalid.

Reply:
Yes, please keep all the validation code. In general I think it's good practice to set `type` to `number` if you're only accepting numbers, that's all.